vme/src/dil.h

```diff
@@ -15,13 +15,63 @@
 
 #define SKIP 0xffffffff /* skip label/index defined */
 
+enum DilVarType_e : uint8_t
+{
+    /* Legal variable values */
+    DILV_UP=1,  /* unit pointer Rexpr Var */
+    DILV_SP=2,  /* string pointer Rexpr Var */
+    DILV_SLP=3, /* string list pointer Var */
+    DILV_EDP=4, /* extra description pointer Rexpr Var */
+    DILV_INT=5, /* integer Rexpr Var */
+
+    /* Other return values */
+    DILV_UPR=6,  /* unit pointer ref Lexpr */
+    DILV_SPR=7,  /* string pointer Rexpr */
+    DILV_SLPR=8, /* string list pointer */
+    DILV_EDPR=9, /* extra descr pointer ref Lexpr */
+
+    /* Different pointers for different size fields */
+    DILV_SINT1R=10, /* integer ref, 8 bit, sgn Lexpr */
+    DILV_SINT2R=11, /* integer ref, 16 bit, sgn Lexpr */
+    DILV_SINT4R=12, /* integer ref, 32 bit, sgn Lexpr */
+    DILV_UINT1R=13, /* integer ref, 8 bit, usgn Lexpr */
+    DILV_UINT2R=14, /* integer ref, 16 bit, usgn Lexpr */
+    DILV_UINT4R=15, /* integer ref, 32 bit, usgn Lexpr */
+
+    DILV_ERR=16,  /* fatal error value */
+    DILV_NULL=17, /* value is null */
+    DILV_FAIL=18, /* value is failed */
+
+    DILV_UEDP=19,    /* FOR COMPILER ONLY! (shared fields) */
+    DILV_FUNC=20,    /* FOR COMPILER ONLY! functions */
+    DILV_PROC=21,    /* FOR COMPILER ONLY! procedures */
+    DILV_HASHSTR=22, /* Hashed String */
+
+    DILV_ZP=23,     /* Zone Pointer Rexpr */
+    DILV_ZPR=24,    /* zone pointer Lexpr */
+    DILV_UZP=25,    /* FOR COMPILER ONLY! unit/zone shared */
+    DILV_EZP=26,    /* FOR COMPILER ONLY! extra/zone shared */
+    DILV_UEZP=27,   /* FOR COMPILER ONLY! all shared */
+    DILV_SP_SLP=28, /* FOR COMPILER ONLY! all shared */
+    DILV_CP=29,     /* Command Pointer Rexpr */
+    DILV_CPR=30,    /* Command pointer Lexpr */
+    DILV_UEZCP=31,  /* FOR COMPILER ONLY! all shared */
+    DILV_UCP=32,    /* FOR COMPILER ONLY! all shared */
+    DILV_ESLP=33,   /* FOR COMPILER ONLY! all shared */
+    DILV_ILP=34,    /* intlist pointer Rexpr */
+    DILV_ILPR=35,   /* intlist pointer Lexpr */
+    DILV_ESLIP=36,  /* For compiler shared e sl il */
+    DILV_ALL=37    /* sym functions can return anything at runtim */
+};
+#define DILV_MAX 37    // Max DILV_ number, used in array allocation
+
 /* The following "dilarg" structs are used only for VMC purposes to allow
    the user to enter the : dilcopy prg@zon(arg, arg, arg); instead of the
    stupid string format */
 
 struct dilargtype
 {
-    ubit8 type;
+    DilVarType_e type;
```

Comment:
Maybe add a INVALID type value that the enums can default too. Zero seems good, didn't see it used above